### PR TITLE
feat: added the husky to the project.

### LINKED ISSUE #78: [Feature Request]: Add the husky to the project
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Feature Description

currently we don't have husky in our project if we add this one the code will automatically formatted when someone try to commit their code This is easy to review the code for the maintainer at the same time we can maintain the code fromat.

### Record

- [X] I agree to follow this project's Code of Conduct
- [X] I want to work on this issue
- [X] I'm willing to provide further clarification or assistance if needed.

### PR DESCRIPTION
**Pull request by Ameerjafar**
- Fixes #78 

## I have add the husky for the project.

https://github.com/user-attachments/assets/b0a1ff1a-425c-4575-b9d0-1dee3a76308b

## ✅ Checklist

Before submitting the PR, please make sure you have completed the following:

- [yes ] My code follows the style guidelines of this project.
- [ yes ] Read Contributing.md

## 🏷️ Types of Changes

What type of changes does your code introduce? (Check all that apply)

- [ ] Bug fix (non-breaking change which fixes an issue) 🐛
- [yes ] New feature (non-breaking change which adds functionality) ✨
- [ ] UI enhancement (non-breaking change which enhances UI) 🎨
- [ ] Documentation update 📚

## Reminder:
Please ensure your pull request is made to the `test` branch, not `main`.

*Thank you for contributing to Talk-Trove! We look forward to reviewing your PR and getting it merged!*

### CODE REVIEW

**Comment by Ameerjafar** [comment]
@kartikay-bigsteps can you review the pr

**Comment by Ameerjafar** [comment]
@Sibam-Paul can you kindly review the pr

**Comment by Sibam-Paul** [comment]
@Ameerjafar come to discord

**Comment by Ameerjafar** [comment]
@Sibam-Paul can I come discord now

File: package.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "scripts": {
+    "prepare": "huksy install"
+  },
+  "devDependencies": {
+    "husky": "^8.0.0"
+  },
+  "dependencies": {
+    "lint-staged": "^15.2.10"
+  }
```
Comment (Sibam-Paul) [inline package.json:10]: Do we need this package.json here??

**Review by Sibam-Paul — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (Ameerjafar) [reply package.json:10]: yes we need because we can only initialize the .husky folder in the root repo

**Comment by Ameerjafar** [comment]
@Sibam-Paul @kartikayasija kindly review it

**Comment by Ameerjafar** [comment]
@kartikayasija  @Sibam-Paul  can you kindly review the pr.

**Pull request closed without merging**